packages/components/src/components/dialog/dialog.tsx

```diff
@@ -68,9 +89,9 @@ const Dialog = ({
     };
 
     const handleClose = () => {
-        if (onClose) {
+        if (typeof onClose === 'function') {
```

Comment:
why just not to have here onClose?.() and onCancel?.() instead of these if's ...? please consider it :)

Reply:
I believe there is no sense  to check if onClose or onCancel are functions, because you set them as a functions in types.  so if this function is not passes via props, this param is undefined. 
so you can check this simply if (onClose) ....
how do you think?